### ISSUE TITLE
Better failure output

### ISSUE THREAD

**Issue by innovate-invent** [hashicorp/terraform#26820]
### Current Terraform Version

```
0.13.5
```

### Use-cases
In the event of a provider panic, terraform should provide the user with a more detailed error output than:
```
Error: rpc error: code = Unavailable desc = transport is closing
```

### Attempted Solutions
To get a meaningful error, you need to set TF_LOG=DEBUG and sift through a terribly long log.

### Proposal
Terraform should output which provider failed and possibly dump the providers exception to a file for recovery. Running terraform again to attempt to recover the exception after enabling the log may not reproduce the exception.

**Comment by richardj-bsquare** [comment]
I have a third-party provider which is very rudimentary, and it outputs this when it fails:

`Error:`

Not at all helpful after 500 lines of refreshing state statements.

**Comment by jbardin** [comment]
Hi @innovate-invent 

The content of the grpc errors, as well as provider panics have been addressed by #26694. 

Thanks!
